pkg/kube/inject/webhook.go

```diff
@@ -894,6 +894,8 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 	var proxyUID *int64
 	var proxyGID *int64
 	if hasOnlyIstioProxyContainer(pod) {
+		// we're injecting a gateway pod
+		// we set proxyUID/GID to whatever UID/GID is specified in securityContext.RunAsUser/RunAsGroup
```

Comment:
does the loop below (`for _, c := range pod.Spec.Containers {`) make any sense since this pod contains only the proxy container?

Reply:
I think it does. Nothing blocks you to deploy a sidecar container - I don't mean istio-proxy in this case. A user might deploy a sidecar container for collecting logs or any other custom component.

Reply:
This if statement was changed in https://github.com/maistra/istio/pull/615. Before that change, the for loop was necessary, but after that PR, that's no longer the case.

Reply:
Should we clean this up in this PR, a different PR, or just in 2.4?

Reply:
I'd do in this PR, separate commit

Reply:
Done.